QE changes

JMPolman opened the pull request:
Changes made since last feedback

JMPolman committed:
ordening and styling

JMPolman committed:
Form Styling + Random order option

JMPolman committed:
removed error cause

JMPolman committed:
Merge branch 'main' of https://github.com/JMPolman/qe-app into main

JMPolman committed:
started checking for answers, changed radio button props

JMPolman committed:
Answer callbacks

JMPolman committed:
Update Answers instead of adding to array

JMPolman committed:
creates an array with empty object according to number of questions.

JMPolman committed:
New Features

-Added Showing limited questions per 'page'
-Added current question number
-Added progressbar progression

JMPolman committed:
New features

Added progress transition animation

JMPolman committed:
Minor fixed

-Progressbar to 100% on last question
-change text of button on last page

JMPolman committed:
Update QuestionnaireRenderer.js

frbl commented on comps/form.js:
```diff
@@ -6,29 +6,43 @@ function JSONForm(props){
 		setFormVisibility(false)
 	}
 
+	function handleChange(){
+		if (random) {
+			setRandom(false)
+			console.log(random)
+		} else{
+			setRandom(true)
+			console.log(random)
+		}
+	}
```
ah, perhaps a good name would be `toggleRandom`?

frbl commented on comps/navbar.js:
```diff
@@ -1,9 +1,26 @@
-function Navbar(props){
+function Navbar({q, currentQ}){
+
+	const qID = currentQ + 1
```
try not to use magic numbers, but instead use constants for those. Also extract this to a function

frbl commented on comps/navbar.js:
```diff
@@ -1,9 +1,26 @@
+function Navbar({q, currentQ}){
+
+	const qID = currentQ + 1
+	const stepSize = 12 / q
```
12?

frbl commented on comps/navbar.js:
```diff
@@ -14,15 +31,15 @@ function Navbar(props){
-	        		<div className="inline-flex bg-gradient-to-r from-UMOgreen via-UMOblue to-UMOpurple mr-5 mt-2 h-2 w-8/12 rounded-md">
-	        			<div className="w-5/12 rounded-l-md h-2">
+	        		<div className="inline-flex justify-end bg-gradient-to-r from-UMOgreen via-UMOblue to-UMOpurple mr-5 h-2 w-8/12 rounded-md mt-2">
+	        			<div className={"w-" +(full ? "0" : calc) +" bg-light h-full clipwacsd rounded-r-md transition-width ease-in duration-300"}>
```
watch out here. This class might be removed by the PostCSS optimizer (the `w-0` or `w-XX` ones)

frbl commented on comps/navbar.js:
```diff
@@ -14,15 +31,15 @@ function Navbar(props){
 	        		</div>
 
-	        		<p>Vraag x / {props.q}</p>
+	        		<p>Vraag {currentQ + 1} / {q}</p>
```
i18n (although thats probably out of scope)

frbl commented on comps/range.js:
```diff
@@ -1,27 +1,27 @@
+	const rangeSteps = labels;
 
-	const stepDistance = (props.max - props.min) / (rangeSteps.length-1);
+	const stepDistance = (max - min) / (rangeSteps.length-1);
```
use a function here to make clear what you're doing

frbl commented on comps/submitbtn.js:
```diff
@@ -0,0 +1,22 @@
+
+function SubmitBtn({submitAnswers, max, first, last, list, id}){
+
+	var text = "Volgende"
+
+	if (last === max){
+		text = "Inleveren"
+	}
```
```suggestion
	const buttonText = last === max ? "Inleveren" : "Volgende"
```
Try to prefer `const` over `let` over `var` (actually, try never to use var)

frbl commented on comps/submitbtn.js:
```diff
@@ -0,0 +1,22 @@
+	return(
+
+		<div className="my-28"> 
+			{button}
```
```suggestion
			<button className="block mx-auto px-10 py-3 rounded-lg bg-UMOblue text-primary hover:bg-UMOpurple transition ease-in duration-200" onClick={(event) => submitAnswers(max, first, last, list)} >{text}</button>
```

frbl replied:
also `UMOblue` is not really generic

frbl commented on comps/submitbtn.js:
```diff
@@ -0,0 +1,22 @@
+	if (last === max){
+		text = "Inleveren"
+	}
+
+	const button = <button className="block mx-auto px-10 py-3 rounded-lg bg-UMOblue text-primary hover:bg-UMOpurple transition ease-in duration-200" onClick={(event) => submitAnswers(max, first, last, list)} >{text}</button>
+
```
```suggestion
```

frbl commented on comps/textarea.js:
```diff
@@ -1,11 +1,11 @@
-			<h1 className="my-5 text-xl w-full font-display text-question">{props.question}</h1>
-			<textarea className="w-full py-1 border-b-2 border-blue  focus:outline-none"></textarea>
+		<div id={id}>
+			<h1 ref={refAnchor} className={"my-5 text-xl w-full font-display" + (active ? ' text-primary' : ' text-question')}>{question}</h1>
```
you probably need the refAnchors?

frbl commented on comps/textfield.js:
```diff
@@ -1,12 +1,18 @@
+function TextField({id, active, question, onUpdate, ref}){
+
+	const isActive = active;
+	// const answer = this.value;
+	// console.log(isActive)
```
```suggestion
```

frbl commented on comps/textfield.js:
```diff
@@ -1,12 +1,18 @@
-function TextField(props){
+function TextField({id, active, question, onUpdate, ref}){
+
+	const isActive = active;
```
why not just use `active`? or rename the prop?

frbl commented on tailwind.config.js:
```diff
@@ -20,14 +20,21 @@ module.exports = {
        'secondary': '#00B1E9',
        'danger': '#e3342f',
        'question':'#232323',
+       'UMOgreen': '#3baa34',
```
perhaps extract this to an override css file? so we can change it for each project

frbl replied:
using semantic names is better in that case, so instead of 'green' use something like 'primary color'

JMPolman committed:
Update README.md

frbl committed:
fix: rename files

JMPolman committed:
Merge pull request #4 from JMPolman/feature/fb-fix-capitalization

fix: rename files

JMPolman committed:
Update package.json

JMPolman committed:
Merge branch 'main' of https://github.com/JMPolman/qe-app into main

JMPolman committed:
Update package.json

JMPolman committed:
Update package.json

JMPolman committed:
Update package.json

JMPolman committed:
Update tailwind.config.js

JMPolman committed:
removal home.module.css

JMPolman committed:
Update tailwind.config.js

JMPolman committed:
Update tailwind.config.js

JMPolman committed:
Update Navbar.js

JMPolman committed:
Update Navbar.js

JMPolman committed:
Update Navbar.js

JMPolman committed:
Update tailwind.config.js

JMPolman committed:
Update tailwind.config.js

JMPolman committed:
Minor changes

max-date
placeholder text instead of value
Finished message

JMPolman committed:
Added feature + adjusted feedback

-End of questionnaire message.

JMPolman committed:
Update date.js

fix max date

JMPolman committed:
Major Changes

- Atomic Design File System
- Recursive questionnaire logic
- 2 test questionnaires
- Adaptable styling
- Mobile Layout

JMPolman committed:
postcss grid-cols fix

JMPolman committed:
grid-cols post-css

JMPolman committed:
final grid-cols fix
